src/Angor/Client/Components/MessageComponent.razor

```diff
@@ -0,0 +1,637 @@
+            {
+                await OnNotification.InvokeAsync("Missing required message information");
+                return;
+            }
+
+            // Load messages
+            await LoadMessages();
+
+            // Subscribe to new messages
+            SubscribeToMessages();
+        }
+        catch (Exception ex)
+        {
+            await OnNotification.InvokeAsync($"Failed to initialize messaging: {ex.Message}");
+        }
+        finally
+        {
+            isLoadingMessages = false;
+            StateHasChanged();
+        }
+    }
+
+    private async Task LoadMessages()
+    {
+        if (string.IsNullOrEmpty(CurrentUserPrivateKeyHex) ||
+            string.IsNullOrEmpty(ContactHexPub)) {
+            Logger.LogWarning("LoadMessages - Missing required keys");
+            return;
+        }
+
+        directMessages.Clear();
+
+        try
+        {
+            // Use pubkeys directly without additional formatting
+            string userHexPubkey = string.IsNullOrEmpty(CurrentUserNpub) ? 
+                "" : NostrHelper.ConvertBech32ToHex(CurrentUserNpub);
+                
+            // Load message history - messages sent FROM current user TO contact
+            await _RelayService.LookupDirectMessagesForPubKeyAsync(
+                ContactHexPub,
+                DateTime.UtcNow.AddDays(-7),
+                100,
+                async eventMessage => await ProcessDirectMessage(eventMessage),
+                userHexPubkey
+            );
+            
+            // Load message history - messages sent FROM contact TO current user
+            await _RelayService.LookupDirectMessagesForPubKeyAsync(
+                userHexPubkey,
+                DateTime.UtcNow.AddDays(-7),
+                100,
+                async eventMessage => await ProcessDirectMessage(eventMessage),
+                ContactHexPub
+            );
+
+            // Sort messages by timestamp
+            directMessages = directMessages.OrderBy(m => m.Timestamp).ToList();
+
+            // Scroll to bottom after loading messages
+            await JS.InvokeVoidAsync("scrollToBottom", messagesContainerRef);
+        }
+        catch (Exception ex)
+        {
+            Logger.LogError($"Failed to load messages: {ex.Message}");
+            await OnNotification.InvokeAsync($"Failed to load messages: {ex.Message}");
+        }
+    }
+
+    private void SubscribeToMessages()
+    {
+        try
+        {
+            // Dispose existing subscription if any
+            messageSubscription?.Dispose();
+
+            // Get the client
+            var nostrClient = _nostrCommunicationFactory.GetOrCreateClient(_networkService);
+
+            // Get pubkeys for comparison
+            string currentUserPubKey = NostrHelper.ConvertBech32ToHex(CurrentUserNpub);
+            
+            // Subscribe to incoming messages
+            messageSubscription = nostrClient.Streams.EventStream
```

Comment:
this is way too low lever to keep on the component, this code should be in a dedicated service